### PR TITLE
fix(server): fixes mtls authentication

### PR DESCRIPTION
**Pull request by coltmcnealy-lh**
- properly inspect client certificate to determine principal id
- refactor grpc server interceptors to be more clear
- refactor boundaries between authorization and authentication

### CODE REVIEW

File: server/src/main/java/io/littlehorse/common/LHServerConfig.java
Comment (coltmcnealy-lh) [inline server/src/main/java/io/littlehorse/common/LHServerConfig.java:1]: The `git diff` on this file is not great; I suggest looking at the before and after side-by-side. The change I made was quite simple but `git` is making it look very complex.

File: server/src/main/java/io/littlehorse/server/auth/LHServerInterceptor.java
Comment (coltmcnealy-lh) [inline server/src/main/java/io/littlehorse/server/auth/LHServerInterceptor.java:1]: @sauljabin do you know why we need this file?

File: server/src/main/java/io/littlehorse/server/auth/MTLSServerAuthorizer.java [hunk 1]
```diff
@@ -1,29 +0,0 @@
...
-import io.littlehorse.common.util.CertificateUtil;
-import io.littlehorse.server.listener.MTLSConfig;
-
-public class MTLSServerAuthorizer implements ServerAuthorizer {
-
-    private final String commonName;
-
-    public MTLSServerAuthorizer(MTLSConfig mtlsConfig) {
-        try {
-            this.commonName = CertificateUtil.getCommonNameFromCertificate(mtlsConfig.getCaCertificate());
```
Comment (coltmcnealy-lh) [inline server/src/main/java/io/littlehorse/server/auth/MTLSServerAuthorizer.java:16]: This is where the problem was: this code pulls the Common Name out of the CA Certificate that gets mounted on the LH Server in `LHS_CA_CERT`, and uses that as the principal id for every request, while completely ignoring the incoming request's certificates.

File: server/src/main/java/io/littlehorse/server/auth/RequestSanitizer.java
Comment (coltmcnealy-lh) [inline server/src/main/java/io/littlehorse/server/auth/RequestSanitizer.java:1]: This file is not needed; the `InsecureAuthenticator` overwrites it to `anonymous`. I updated the unit tests.

**Review by Snarr — APPROVED** [review]

File: local-dev/issue-certificates.sh [hunk 2]
```diff
@@ -51,7 +51,7 @@ echo "Creating Client Certificates"
 openssl req -newkey rsa:2048 -nodes \
     -out "$CLIENT_PATH/client.csr" \
     -keyout "$CLIENT_PATH/client.key" \
-    -subj "/CN=localhost/O=client organization" \
+    -subj "/CN=obiwan/O=client organization" \
```
Comment (sauljabin) [inline local-dev/issue-certificates.sh:54]: I know your favorite is Obi-Wan but Anakin is much better (my favorite)

**Review by sauljabin — APPROVED** [review]

File: server/src/main/java/io/littlehorse/server/auth/LHServerInterceptor.java
Comment (sauljabin) [reply server/src/main/java/io/littlehorse/server/auth/LHServerInterceptor.java:1]: interceptor strategy

**Review by sauljabin — APPROVED** [review]

**Review by eduwercamacaro — APPROVED** [review]

**Pull request merged**